src/spec.c

```diff
@@ -3217,3 +3269,62 @@ void Indexes_ReplaceMatchingWithSchemaRules(RedisModuleCtx *ctx, RedisModuleStri
   Indexes_SpecOpsIndexingCtxFree(to_specs);
 }
 ///////////////////////////////////////////////////////////////////////////////////////////////
+
+// Debug Scanner Functions
+
+static DebugIndexesScanner *DebugIndexesScanner_New(StrongRef global_ref) {
+
+  DebugIndexesScanner *dScanner = rm_realloc(IndexesScanner_New(global_ref), sizeof(DebugIndexesScanner));
+
+  dScanner->maxDocsTBscanned = globalDebugCtx.bgIndexing.maxDocsTBscanned;
+  dScanner->maxDocsTBscannedPause = globalDebugCtx.bgIndexing.maxDocsTBscannedPause;
+  dScanner->wasPaused = false;
+  dScanner->status = DEBUG_INDEX_SCANNER_CODE_NEW;
+  dScanner->base.isDebug = true;
+
+  IndexSpec *spec = StrongRef_Get(global_ref);
+  spec->scanner = (IndexesScanner*)dScanner;
+
+  return dScanner;
+}
+
+// This function is currently not called as the pointer is freed via the IndexSpec_Free function
+// It is kept here for completeness and in case it is needed in the future
+static void DebugIndexesScanner_Free(DebugIndexesScanner *dScanner)
+{
+  IndexesScanner_Free((IndexesScanner*)dScanner);
+}
```

Comment:
Not covered - is it unused?
```suggestion
static void DebugIndexesScanner_Free(DebugIndexesScanner *dScanner) {
  IndexesScanner_Free((IndexesScanner*)dScanner);
}
```

Reply:
Yes, unused, its in the comments.
Left for completion.